product-scenarios/2-Collaborative-API-Development/2.1-Manage-publisher-portal-operations-for-users/2.1.1-Manage-API-publishing-according-to-the-publisher-roles-and-permission/src/test/java/org/wso2/am/scenario/tests/rest/api/publisherRoleAndPermission/PublishAPIByValidRoleCategoryTestCase.java

```diff
@@ -0,0 +1,250 @@
+        HttpResponse publishAPI = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.PUBLISHED);
+        assertFalse(publishAPI.getData().contains("PUBLISHED"), "API has been created in publisher");
+
+        updateUser(testUser, "internal/publisher", role, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(testUser, password);
+        HttpResponse publisherPublishAPI = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.PUBLISHED);
+        verifyResponse(publisherPublishAPI);
+        assertTrue(publisherPublishAPI.getData().contains("PUBLISHED"),
+                "API has not been created in publisher");
+
+        HttpResponse changeAPIStatus = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.CREATED);
+        verifyResponse(changeAPIStatus);
+        assertTrue(changeAPIStatus.getData().contains("CREATED"),
+                "API has been created in publisher");
+
+        updateUser(testUser, "admin", role, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(testUser, password);
+        HttpResponse adminChangeAPIStatus = changeAPILifeCycleStatus(apiName, creatorUser, APILifeCycleState.PUBLISHED);
+        verifyResponse(adminChangeAPIStatus);
+        assertTrue(adminChangeAPIStatus.getData().contains("PUBLISHED"),
+                "API has not been created in publisher");
+    }
+
+    public void createAPI(APIRequest apiRequest) throws Exception {
+
+        HttpResponse apiCreationResponse = apiPublisher.addAPI(apiRequest);
+        assertEquals(apiCreationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response Code miss matched when creating the API");
+        verifyResponse(apiCreationResponse);
+    }
+
+    public void getAPI(String apiName, String username, String apiVersion) throws Exception {
+
+        HttpResponse apiResponsePublisher = apiPublisher.getAPI(apiName, username, apiVersion);
+        verifyResponse(apiResponsePublisher);
+        assertTrue(apiResponsePublisher.getData().contains(apiName), apiName + " is not visible in publisher");
+        verifyResponse(apiResponsePublisher);
+    }
+
+    public HttpResponse changeAPILifeCycleStatus(String apiName, String username, APILifeCycleState apiLifeCycleState)
+            throws Exception {
+
+        APILifeCycleStateRequest updateRequest =
+                new APILifeCycleStateRequest(apiName, username, apiLifeCycleState);
+        HttpResponse apiResponsePublishAPI = apiPublisher.changeAPILifeCycleStatus(updateRequest);
+        return apiResponsePublishAPI;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+
+        for (Map.Entry<String, String> entry : apiNames.entrySet()) {
+            String apiName = entry.getKey();
+            String provider = entry.getValue();
+            apiPublisher.deleteAPI(apiName, apiVersion, provider);
+            deleteUser(provider, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
```

Comment:
Aren't these users in usersList?

Reply:
They are not in the usersList since I put them into hashMap